### DISCUSSION TITLE
Question about pretrained model

### DISCUSSION THREAD

**Discussion by aminogene** [STVIR/pysot#522] [Q&A]
我想知道backbone的预训练模型（renst50.model）是如何得到的，所用的网络结构是siamrpn++中的resnet50的网络结构吗？如果用siamrpn++中的resnet50进行预训练，renset50最终的fc层的输入通道是多少？在输入fc前的特征图尺寸是多少？是否需要对此时的特征图尺寸进行压缩？

**Comment by shiyi-mu** [comment]
我也遇到同样问题，论文使用的resnet50和原始resnet50的stride设置不同，导致layer4的输出尺寸过大，再过最后平均池化之后变形成B*C*1*1和原始resnet50此处的通道数C不同，我认为不能使用原始版本Resnet50的池化设置和全连接分类头，我尝试了直接修改最后池化的stride和全连接输入通道，效果不理想，希望项目作者能够开源resnet50预训练的部分代码。

**Comment by ZhiyuanChen** [comment]
论文中应该有提到，预训练阶段使用的是标准resnet设置。

**Reply by aminogene** [reply]
好的，谢谢你

**Reply by aminogene** [reply]
请问resnet50是先进行结构修改（根据论文中提到的“we reduce the effective strides at the last two block from 16 pixels and 32 pixels to 8 pixels bymodifying the conv4 and conv5 block to have unit spatial stride, and also increase its receptive field by dilated convolutions. An extra 1 × 1 convolution layer is appended to each of block outputs to reduce the channel to 256.”），然后再用Imagenet数据训练得到backbone；还是，用标准的resnet50采用Imagenet数据训练得到backbone，在训练siamRPN++的时候对backbone的结构强制进行调整？

**Reply by ZhiyuanChen** [reply]
用标准的resnet50采用Imagenet数据训练得到backbone，在训练siamRPN++的时候对backbone的结构强制进行调整

**Reply by aminogene** [reply]
谢谢您的回复！我们尝试加载了标准resnet50（官方）在Imagenet数据集上训练得到的backbone（不是你们提供的backbone），运行RPN++训练时报错。请问我们该做哪些调整？谢谢您

**Reply by ZhiyuanChen** [reply]
所以报错是啥呢……

**Reply by aminogene** [reply]
报错在于：如果用标准resnet50的训练得到的参数模型作为预训练模型加载入siamrpn++网络中，在layer2,3,4中的downsample处，由于kernel大小不一致导致了报错

**Reply by ZhiyuanChen** [reply]
> 报错在于：如果用标准resnet50的训练得到的参数模型作为预训练模型加载入siamrpn++网络中，在layer2,3,4中的downsample处，由于kernel大小不一致导致了报错

down sample怎么会有kernel，你们用的什么downsample呀

**Reply by aminogene** [reply]
resnet50跳跃连接部分downsample的kernel

**Reply by aminogene** [reply]
这是将标准resnet50的预训练模型作为backbone加载到siamRPN++网络，对siamRPN++网络进行训练时的报错：
![image](https://user-images.githubusercontent.com/30854908/117395504-a3693380-af2a-11eb-9f97-65f441ec731e.png)
在make_layer的downsample处，由于siamRPN++网络中定义的kernel与加载的标准resnet50的预训练模型的kernel大小不一致，导致报错
